scvi/model/_lda.py

```diff
@@ -0,0 +1,183 @@
+
+    Parameters
+    ----------
+    adata
+        AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    n_components
+        Number of components/topics to model.
+    n_hidden
+        Number of nodes in the hidden layer of the encoder.
+    cell_component_prior
+        Prior of cell component distribution. If `None`, defaults to `1 / n_components`.
+    component_gene_prior
+        Prior of component gene distribution. If `None`, defaults to `1 / n_components`.
+    """
+
+    def __init__(
+        self,
+        adata: AnnData,
+        n_components: int = 20,
+        n_hidden: int = 128,
+        cell_component_prior: Optional[float] = None,
+        component_gene_prior: Optional[float] = None,
+    ):
+        # in case any other model was created before that shares the same parameter names.
+        pyro.clear_param_store()
+
+        super().__init__(adata)
+
+        self.module = LDAPyroModule(
+            n_input=self.summary_stats["n_vars"],
+            n_components=n_components,
+            n_hidden=n_hidden,
+            cell_component_prior=cell_component_prior,
+            component_gene_prior=component_gene_prior,
+        )
+
+    def _check_var_equality(self, adata: AnnData):
+        """
+        Preliminary check for var name equality between AnnData objects.
+
+        Checks that the var names are equivalent between the source AnnData object and
+        the AnnData passed in. Raises a ValueError if not.
+
+        Parameters
+        ----------
+        adata
+            AnnData to compare against.
+        """
+        source_var_names = self.adata.var_names.astype(str)
+        user_var_names = adata.var_names.astype(str)
+        if not np.array_equal(user_var_names, source_var_names):
+            raise ValueError(
+                "`adata` passed into `transform` does not have matching var_names "
+                "with the source adata the model was trained with."
+            )
+
+    def _check_if_not_trained(self):
+        """Check if the model is not trained. Raises a ValueError if not."""
+        if not self.is_trained_:
+            raise ValueError(
+                "Trying to query inferred values from an untrained model. Please train the model first."
+            )
+
+    def get_components(self) -> pd.DataFrame:
+        """
+        Gets the component to gene transition matrix.
+
+        Parameters
+        ----------
+        adata
+            AnnData to transform. If None, returns the component to gene transition matrix for
+            the source AnnData.
+
+        Returns
+        -------
+        A `n_components x n_var` Pandas DataFrame containing the component to gene transition matrix.
+        """
+        self._check_if_not_trained()
+
+        return pd.DataFrame(
+            data=self.module.components.numpy(), columns=self.adata.var_names
+        )
+
+    def transform(self, adata: Optional[AnnData] = None) -> pd.DataFrame:
```

Comment:
```suggestion
    def get_latent_representation(self, adata: Optional[AnnData] = None) -> pd.DataFrame:
```

This is just to be consistent with the vae-based models

Reply:
this also needs all the same params as the vaemixin get_latent... (e.g., batch size, etc)